prereise/gather/griddata/hifld/data_access/load.py

```diff
@@ -186,6 +189,104 @@ def get_hifld_electric_substations(path):
     )
 
 
+def _join_line_segments(segments, threshold=0.001):
+    """Given a list of line segments, join them together as best as possible.
+
+    :param list segments: list of segments, each defined as a list of (lon, lat)
+        coordinate pairs.
+    :param float threshold: minimum angular distance for a segment to be considered.
+        Defined as the angular distance from the segment start to segment end (degrees).
+    :return: (*list*) -- consolidated list of points.
+    """
+
+    def _generate_linear_spanning_tree(segments):
+        """Given a list of segments, create a minimum spanning tree and ensure that it
+        is linear (i.e. there are two nodes of degree 1, all other nodes are degree 2).
+
+        :param list segments: list of segments, each defined as a list of (lon, lat)
+            coordinate pairs.
+        :return: (*networkx.Graph*) -- a minimum spanning tree from the segments.
+        :raises ValueError: if the minimum spanning tree is not linear
+        """
+        # Build a Graph representing the existing connections between segment endpoints
+        g = nx.Graph()
+        # add a 'start' and 'end' node for each segment, and a zero-cost edge b/w them
+        for i in range(len(segments)):
+            (lon1, lat1), (lon2, lat2) = segments[i][0], segments[i][-1]
+            segment_angular_distance = angular_distance(
+                latlon_to_xyz(lat1, lon1), latlon_to_xyz(lat2, lon2)
+            )
+            if segment_angular_distance < threshold:
+                # Segment is too small, so we won't add this to the graph
+                continue
+            g.add_nodes_from([(i, "start"), (i, "end")])
+            g.add_edge((i, "start"), (i, "end"), weight=0)
```

Comment:
Looking at the documentation of `add_edge`, it seems that the nodes will be automatically added if they are not already in the graph. So I guess we can discard the previous line.

Reply:
Done.